mainnet-contracts/src/PufferVaultV3.sol

```diff
@@ -0,0 +1,217 @@
+     * @dev See {IERC4626-totalAssets}. pufETH, the shares of the vault, will be backed primarily by the WETH asset.
+     * However, at any point in time, the full backings may be a combination of stETH, WETH, Eigenpod ETH Rewards and ETH.
+     * `totalAssets()` is calculated by summing the following:
+     * - WETH held in the vault contract
+     * - ETH held in the vault contract
+     * - PUFFER_ORACLE.getLockedEthAmount(), which is the oracle-reported Puffer validator ETH locked in the Beacon chain
+     * - stETH held in the vault contract, in EigenLayer's stETH strategy, and in Lido's withdrawal queue. (we assume stETH is always 1:1 with ETH since it's rebasing)
+     * - ETH held in the eigenpods as a result of receiving rewards.
+     * NOTE on the native ETH deposits:
+     * When dealing with NATIVE ETH deposits, we need to deduct callvalue from the balance.
+     * The contract calculates the amount of shares (pufETH) to mint based on the total assets.
+     * When a user sends ETH, the msg.value is immediately added to address(this).balance.
+     * Since address(this.balance)` is used in calculating `totalAssets()`, we must deduct the `callvalue()` from the balance to prevent the user from minting excess shares.
+     * `msg.value` cannot be accessed from a view function, so we use assembly to get the callvalue.
+     * @return The total assets held by the vault.
+     */
+    function totalAssets() public view virtual override returns (uint256) {
+        uint256 callValue;
+        // solhint-disable-next-line no-inline-assembly
+        assembly {
+            callValue := callvalue()
+        }
+        return _ST_ETH.balanceOf(address(this)) + getPendingLidoETHAmount() + getELBackingEthAmount()
+            + _WETH.balanceOf(address(this)) + (address(this).balance - callValue) + PUFFER_ORACLE.getLockedEthAmount()
+            + getTotalRewardMintAmount();
+    }
+
+    /**
+     * @notice Returns the total reward mint amount.
+     * @return The total reward mint amount.
+     */
+    function getTotalRewardMintAmount() internal view returns (uint256) {
+        VaultStorage storage $ = _getPufferVaultStorage();
+        return $.totalRewardMintAmount;
+    }
+
+    /**
+     * @notice Mints and bridges rewards according to the provided parameters.
+     * @param params The parameters for bridging rewards.
+     */
+    function mintAndBridgeRewards(MintAndBridgeParams calldata params) external payable restricted {
+        VaultStorage storage $ = _getPufferVaultStorage();
+
+        if (params.rewardsAmount > $.allowedRewardMintAmount) {
+            revert InvalidMintAmount();
+        }
+
+        if (($.lastRewardMintTimestamp + $.allowedRewardMintFrequency) > block.timestamp) {
+            revert NotAllowedMintFrequency();
+        }
+
+        $.lastRewardMintTimestamp = uint40(block.timestamp);
+        $.totalRewardMintAmount += uint104(params.rewardsAmount);
+
+        _mint(address(this), params.rewardsAmount);
```

Comment:
This is minting pufETH, and it is comming from off-chain. Off chain can't predict the pufETH exchange rate.
Our offchain is calculating stuff in ETH and not in pufETH.